drivers/interrupt_controller/intc_nrf_egu.c

```diff
@@ -0,0 +1,193 @@
+	nrfx_egu_int_enable(&config->egu, 1U << channel);
+
+	return 0;
+}
+
+static int channel_callback_clear(const struct device *dev, int channel)
+{
+	struct egu_channel_data *channel_data = get_channel_data(dev, channel);
+	const struct egu_config *config = dev->config;
+
+	nrfx_egu_int_disable(&config->egu, 1U << channel);
+	channel_data->callback = NULL;
+	channel_data->ctx = NULL;
+
+	return 0;
+}
+
+static int channel_task_trigger(const struct device *dev, int channel)
+{
+	const struct egu_config *config = dev->config;
+
+	nrfx_egu_trigger(&config->egu, channel);
+
+	return 0;
+}
+
+static int channel_alloc(const struct device *dev)
+{
+	struct egu_data *data = dev->data;
+	struct egu_channel_data *channels = data->channels;
+	const struct egu_config *config = dev->config;
+	k_spinlock_key_t key;
+
+	key = k_spin_lock(&data->lock);
+	for (int i = 0; i < config->ch_num; i++) {
+		if (!channels[i].taken) {
+			channels[i].taken = true;
+			return i;
+		}
+	}
+	k_spin_unlock(&data->lock, key);
+
+	return -ENODEV;
+}
+
+static int channel_free(const struct device *dev, int channel)
+{
+	struct egu_data *data = dev->data;
+	struct egu_channel_data *channel_data = get_channel_data(dev, channel);
+	k_spinlock_key_t key;
+
+	egu_channel_callback_clear(dev, channel);
+
+	key = k_spin_lock(&data->lock);
+	channel_data->taken = false;
+	k_spin_unlock(&data->lock, key);
+
+	return 0;
+}
+
+static void egu_event_handler(uint8_t event_idx, void *context)
+{
+	struct egu_data *data = context;
+	const struct device *dev = data->dev;
+	struct egu_channel_data *channel_data = get_channel_data(dev, event_idx);
+
+	if (channel_data->callback) {
+		channel_data->callback(dev, event_idx, channel_data->ctx);
+	}
+}
+
+static struct egu_driver_api nrfx_egu_driver_api = {
+	.channel_callback_set = channel_callback_set,
+	.channel_callback_clear = channel_callback_clear,
+	.channel_task_trigger = channel_task_trigger,
+	.channel_alloc = channel_alloc,
+	.channel_free = channel_free,
+};
+
+static int init_egu(const struct device *dev)
+{
+	struct egu_data *data = dev->data;
+	const struct egu_config *config = dev->config;
+	nrfx_err_t err;
+
+	data->dev = dev;
+	err = nrfx_egu_init(&config->egu, 0, egu_event_handler, data);
+	__ASSERT_NO_MSG(err == NRFX_SUCCESS);
+
+	return 0;
+}
+
+#define INTC_NRF_EGU_DEVICE(idx)								\
+	static struct egu_channel_data egu_##idx##_ch_data[EGU##idx##_CH_NUM];			\
+	static struct egu_data egu_##idx##_data = {						\
+		.channels = egu_##idx##_ch_data,						\
+	};											\
+	static struct egu_config egu_##idx##z_config = {					\
+		.egu = NRFX_EGU_INSTANCE(idx),							\
+		.ch_num = EGU##idx##_CH_NUM,							\
+	};											\
+	static int egu_##idx##_init(const struct device *dev)					\
+	{											\
+		IRQ_CONNECT(DT_IRQN(DT_DRV_INST(idx)), DT_IRQ(DT_DRV_INST(idx), priority),	\
+			    nrfx_isr, nrfx_egu_##idx##_irq_handler, 0);				\
+		return init_egu(dev);								\
+	}											\
+	DEVICE_DT_INST_DEFINE(idx,								\
```

Comment:
Also here you cannot use DT instances.
```suggestion
	DEVICE_DT_DEFINE(EGU(idx),								\
```

Reply:
Removed the use of DT_INST API.